app/javascript/components/close_contact/CloseContact.js

```diff
@@ -86,8 +82,14 @@ class CloseContact extends React.Component {
     let disableCreate = this.state.disableCreate;
     const possiblyRequiredFields = ['first_name', 'last_name', 'primary_telephone', 'email'];
     if (possiblyRequiredFields.includes(event.target.id)) {
-      // This checks that at least one of the possiblyRequiredFields is set to a truthy value
-      disableCreate = !possiblyRequiredFields.some(field => (field === event.target.id ? !!event.target.value : !!this.state[`${field}`]));
+      // Users are required to input at least one name field (first or last) and one Reporting Method Field (phone or email)
+      let hasFirstNameEntry = event.target.id === 'first_name' ? !!event.target.value : !!this.state['first_name'];
+      let hasLastNameEntry = event.target.id === 'last_name' ? !!event.target.value : !!this.state['last_name'];
+      let hasTelephoneEntry = event.target.id === 'primary_telephone' ? !!event.target.value : !!this.state['primary_telephone'];
+      let hasEmailEntry = event.target.id === 'email' ? !!event.target.value : !!this.state['email'];
+      let hasNameEntry = !!hasFirstNameEntry || !!hasLastNameEntry;
+      let hasReportingMethod = !!hasTelephoneEntry || !!hasEmailEntry;
+      disableCreate = !hasNameEntry || !hasReportingMethod;
```

Comment:
This feels like a lot of unnecessary code - can we just have a method that extracts this logic (based on the state rather than the props) and call it in the callback of the `setState` method on line 103 after the state has been updated with the changes? This method would still need to call setState on `disableCreate`.

Reply:
So i do agree. And I initially wrote a way to calculate this in a single loop. But that logic was not easily readable. In some sense, I kinda like how easy to read this is. There's no question what's going on here. But if you think it's over-simplistic, Im happy to improve this logic.

Reply:
I don't find this very easy to read personally. I'm thinking it can be much simpler broken out into a separate function that checks the state in the `setState` callback:
```
        (this.state.close_contact.first_name || this.props.close_contact.last_name) &&
        (this.state.close_contact.primary_telephone || this.stateclose_contact.email)
```

Reply:
I hadn't considered doing it this way. Misunderstood your original comment. Good idea!!